### ISSUE TITLE
LGTM.com - false positive for SolidJS's `for` attribute

### ISSUE THREAD

**Issue by xfix** [github/codeql#9792]
**Description of the false positive**

SolidJS has a `ref` attribute which assigns the element to a variable, see https://www.solidjs.com/docs/latest/api#ref.

I don't think that's simple to support because it is a non-standard extension to JavaScript language.

**URL to the alert on the project page on LGTM.com**

https://lgtm.com/projects/g/pastebinrun/pastebinrun/snapshot/95688ef349b2f4c1f75c116216706d05846da2b3/files/js/views/TextAreaEditor/index.tsx?sort=name&dir=ASC&mode=heatmap#x1dce09b117a9e254:1

**Comment by aeisenberg** [comment]
I have never worked with solid-js, but it does look like `textarea` is never initialized. It's a local variable, so it cannot be accessed outside of the constructor/function it is declared in. Is there some kind of solid-js magic going on here that can set the value of the textarea?

**Comment by xfix** [comment]
Correct. SolidJS uses a special JSX transformation that special-cases `ref`: https://github.com/ryansolid/dom-expressions/tree/main/packages/babel-plugin-jsx-dom-expressions#ref

That said, there is a workaround of using `ref={e => variable = e}` instead, so that's what I have done to deal with this issue. Still, figured I would report this issue.

**Comment by aeisenberg** [comment]
Thanks for the explanation. This does indeed look like a false positive, but capturing it will require embedding analysis rules about solid-js into our core library. This would require some significant engineering effort. I am going to put this on our backlog and if we see more users request solid-js support, we will prioritize this appropriately.
